src/store/modules/toolbar/mutations.ts

```diff
@@ -0,0 +1,28 @@
+import { MutationType, AddRightButtonPayload, ToolbarState, SetRightButtonsPayload, ClearRightButtonsPayload } from '@/store/modules/toolbar/types';
+import { MutationHandlers } from '@/store/types';
+import logger from '@/utils/logger';
+
+const mutations: MutationHandlers<ToolbarState> = {
+
+  [MutationType.ADD_RIGHT_BUTTON] (state: ToolbarState, { button, position }: AddRightButtonPayload) {
+    if (position < 0 || position > state.rightButtons.length) {
```

Comment:
The idea of this error is that if a user is ever calling this mutation and it enters that block, it was a developer error. I think we will be catching all the errors using the logger... which I just realized the logger is not setup correctly at all because this isn't node.

The reason it's a developer error is the fact that you have access to the state when you call the mutation, so it's essentially an IndexOutOfRange